src/cmd/ksh93/tests/emacs.exp

```diff
@@ -0,0 +1,422 @@
+# Check emacs keybindings
+set pid [spawn $ksh]
+expect_prompt
+
+# ==========
+# Enable emacs keybindings
+send "set -o emacs\r"
+expect_prompt
+
+# ==========
+# Populate the command history with a couple of commands.
+send "echo hello\r"
+expect "\r\nhello\r\n" {
+    puts "echo 1 works"
+}
+expect_prompt
+send "echo buh bye\r"
+expect "\r\nbuh bye\r\n" {
+    puts "echo 2 works"
+}
+expect_prompt
+
+# ==========
+# Send ctrl-p to retrieve previous command
+log_test_entry
+send [ctrl P]
+send [ctrl L]
```

Comment:
Why are you sending `ctrl-L` here ?

Reply:
We can probably wrap this control sequence in a procedure called `redraw_prompt` for better readability.

Reply:
Okay. Will do.